### PR TITLE
Use ngrx mock store

### LINKED ISSUE #1357: Implement TestStore
Implement the TestStore, update all the tests that use it and fix the menu tests.
See for info `https://github.com/ngrx/platform/issues/915`

### PR DESCRIPTION
**Pull request by wowshakhov**
closes #1357

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
     "@ngrx/effects": "6.1.0",
     "@ngrx/entity": "6.1.0",
     "@ngrx/router-store": "6.1.0",
-    "@ngrx/store": "6.1.0",
+    "@ngrx/store": "^8.6.0",
```
Comment (andrewbents) [inline package.json]: any breaking changes?

File: src/app/auth/login.component.ts [hunk 2]
```diff
@@ -40,7 +41,7 @@ export class LoginComponent implements OnInit {
     const domainFromQueryParams = this.route.snapshot.queryParams['domain'];
     this.store
       .pipe(
-        select(configSelectors.get('defaultDomain')),
+        select(getDefaultDomain),
```
Comment (andrewbents) [inline src/app/auth/login.component.ts]: do we need to test that?

File: package.json (hunk 1 above)
Comment (wowshakhov) [reply package.json]: The changes suggested in https://github.com/bwsw/cloudstack-ui/issues/1357 require `@ngrx/store/testing` but it's missing in the 6th version.

File: src/app/auth/login.component.ts (hunk 2 above)
Comment (wowshakhov) [reply src/app/auth/login.component.ts]: This is just an extracted constant, no need to change tests

**Pull request merged**